### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by andrey-churkin**
The `create_compression_algorithm_builder()` method never returns `None` because the `extract_compression_algorithm_configs()` contains this check:

https://github.com/openvinotoolkit/nncf/blob/138a7dc4c87e8436b62509abfd14050f3d94f559/nncf/config/extractors.py#L47-L48

### CODE REVIEW

File: nncf/tensorflow/helpers/model_creation.py [hunk 1]
```diff
@@ -74,8 +73,6 @@ def create_compressed_model(model: tf.keras.Model,
             original_model_accuracy = evaluation_args.eval_fn(model)
 
     builder = create_compression_algorithm_builder(config, should_init)
-    if builder is None:
-        return None, model
```
Comment (alexsu52) [inline nncf/tensorflow/helpers/model_creation.py:78]: @andrey-churkin Please fix also the docstring and type hint for return parameters.

Comment (andrey-churkin) [reply nncf/tensorflow/helpers/model_creation.py:78]: Done

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by andrey-churkin** [comment]
Jenkins please stop all builds

**Comment by andrey-churkin** [comment]
Jenkins please retry a build

**Comment by andrey-churkin** [comment]
Jenkins please retry a build

**Review by alexsu52 — APPROVED** [review]

**Comment by alexsu52** [comment]
Jenkins please retry a build

**Pull request merged**